### PR TITLE
Object: improvements to PHP8

### PR DESCRIPTION
**Pull request by daniwe4**

### CODE REVIEW

File: Modules/BookingManager/classes/class.ilObjBookingPool.php [hunk 1]
```diff
@@ -200,7 +200,7 @@ public function delete() : bool
         return true;
     }
     
-    public function cloneObject($a_target_id, $a_copy_id = 0, $a_omit_tree = false)
+    public function cloneObject(int $a_target_id, int $a_copy_id = 0, $a_omit_tree = false) : ?ilObject
```
Comment (utesche) [inline Modules/BookingManager/classes/class.ilObjBookingPool.php:203]: ...no datatype 'bool' for '$a_omit_tree' yet

Comment (daniwe4) [reply Modules/BookingManager/classes/class.ilObjBookingPool.php:203]: Hello @utesche , that's probably lost in the amount of files. Thank you very much for your hint. I'll fix that in an extra PR. Greetings Daniel.

Comment (daniwe4) [reply Modules/BookingManager/classes/class.ilObjBookingPool.php:203]: You will find the PR with the fix here: https://github.com/ILIAS-eLearning/ILIAS/pull/4113

File: Modules/Course/classes/class.ilObjCourse.php [hunk 2]
```diff
@@ -593,16 +593,21 @@ public static function _registrationEnabled(int $a_obj_id) : bool
...
-    public function read()
+    public function read() : void
     {
         parent::read();
         $this->setOrderType(ilContainerSortingSettings::_lookupSortMode($this->getId()));
         $this->__readSettings();
     }
 
-    public function create($a_upload = false)
+    public function create($a_upload = false) : int
```
Comment (utesche) [inline Modules/Course/classes/class.ilObjCourse.php:603]: stumbled upon another one:  argument $a_upload without type (bool) in function class.ilObjCourse::create()..

Comment (daniwe4) [reply Modules/Course/classes/class.ilObjCourse.php:603]: Hi @utesche ,
the parent classes of ilObjCourse doesn't use this parameter. So this is only used by ilObjCourse and doesn't fit into this PR.
But thanks again for the detailed look.

**Pull request merged**